src/loaders/lottie/tvgLottieParser.cpp

```diff
@@ -992,9 +985,16 @@ LottieObject* LottieParser::parseAsset()
         else if (KEY_AS("w")) width = getFloat();
         else if (KEY_AS("h")) height = getFloat();
         else if (KEY_AS("e")) embedded = getInt();
+        else if (KEY_AS("sid")) {
+            obj = new LottieImage;
```

Comment:
The code seems to be deteriorating, as sid relies solely on the image property. The asset can be either a precomp layer or an image. This will be quickly broken when 982/988 lines are both triggered.

Reply:
I changed the logic so that it occurs inside the `if (data) {}` condition (image case)